File: libraryProject_Python/Video.py
from Media import Media

class Video(Media):

	def __init__(self,callNo,title,subjects,description,distributor,notes,series,label):
		self.callNo = callNo
		self.title = title
		self.subjects = subjects
		self.description = description
		self.distributor = distributor
		self.notes = notes
		self.series = series
		self.label = label
		super(Video,self).__init__(self.callNo,self.title,self.subjects,self.notes)


	
	def compare_Other(self,other):
		if other in self.description:
			return True
		elif other in self.distributor:
			return True
		elif other in self.notes:
			return True
		else:
			return False




	def display(self):
		super(Video,self).display()
		print ("description: %s" % (self.description))
		print ("distributor: %s" % (self.distributor))
		print ("series: %s" % (self.series))
		print ("label: %s" % (self.label))
		
		

File: LibraryProject_C++/Video.cpp

#include "Video.h"


Video::Video(const string& call_no,const string& title,const string& subjects,const string& description,const string& distributor,const string& notes,const string& series,const string& label):Media(call_no,title,subjects,notes), description(description),distributor(distributor),series(series),label(label)
{
}


Video::~Video(){}

/*
   compare_other finds if the input string exists in Description, Notes
   and Distributor. It returns boolean value.
 */
bool Video::compare_other(const string& other_string) const{

	if(description.find(other_string)!=string::npos){
		return true;
	}
	else if(Media::compare_other(other_string)){
		return true;
	}
	else if(distributor.find(other_string)!=string::npos){
		return true;
	}
	else{
		return false;
	}
}

/*
   display function to print Video attributes. This function gives
   call to display() of Media class.
 */
void Video::display() const
{
	cout << "------Video------" << endl;

	Media::display();  

	cout<<"Description: "<<description<<endl;
	cout<<"Distributor: "<<distributor<<endl;
	cout<<"Series:      "<<series<<endl;  
	cout<<"Label:       "<<label<<endl<<endl;
}

